src/posix/client.cpp

```diff
@@ -108,14 +132,68 @@ static bool DoWrite(int aFile, const void *aBuffer, size_t aSize)
     return ret;
 }
 
+static void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
+{
+    fprintf(aStream,
+            "Syntax:\n"
+            "    %s [Options]\n"
+            "Options:\n"
+            "    -h  --help                    Display this usage information.\n"
+            "    -I  --interface-name name     Thread network interface name.\n",
+            aProgramName);
+    exit(aExitCode);
+}
+
+static void ParseArg(int aArgCount, char *aArgVector[], ClientConfig *aConfig)
+{
+    memset(aConfig, 0, sizeof(*aConfig));
+
+    optind = 1;
+
+    while (true)
+    {
+        int index  = 0;
+        int option = getopt_long(aArgCount, aArgVector, "hI:", kClientOptions, &index);
+
+        if (option == -1)
+        {
+            break;
+        }
+
+        switch (option)
+        {
+        case OT_POSIX_OPT_HELP:
+            PrintUsage(aArgVector[0], stdout, OT_EXIT_SUCCESS);
+            break;
+        case OT_POSIX_OPT_INTERFACE_NAME:
+            aConfig->mInterfaceName = optarg;
+            break;
+        default:
+            PrintUsage(aArgVector[0], stderr, 128);
```

Comment:
also, I think we should break here instead of printing errors, otherwise, users must issue commands like `ot-ctl -I wpan0 -- XX XX` if OpenThread commands has `-` style options.

Reply:
exit code 128 is actually not magic. It's also used in `script/check-size`.

Reply:
Do you mean we should break without `PrintUsage`? 
Why do you think we have to use `ot-ctl -I wpan0 -- XX XX`. I have tested `ot-ctl -I wpan0 Xx Xx` and it works well.

Reply:
Changed the `shortopts` to `"+hI:"` to address this issue.